nifi-nar-bundles/nifi-media-bundle/nifi-media-processors/src/main/java/org/apache/nifi/processors/media/ExtractMediaMetadata.java

```diff
@@ -0,0 +1,311 @@
+            .build();
+
+    private Set<Relationship> relationships;
+    private List<PropertyDescriptor> properties;
+
+    private final AtomicReference<Pattern> mimeTypeFilterRef = new AtomicReference<>();
+    private final AtomicReference<Pattern> metadataKeyFilterRef = new AtomicReference<>();
+
+    private volatile AutoDetectParser autoDetectParser;
+
+    @Override
+    protected void init(final ProcessorInitializationContext context) {
+
+        final List<PropertyDescriptor> properties = new ArrayList<>();
+        properties.add(MAX_NUMBER_OF_ATTRIBUTES);
+        properties.add(MAX_ATTRIBUTE_LENGTH);
+        properties.add(MIME_TYPE_FILTER);
+        properties.add(METADATA_KEY_FILTER);
+        properties.add(METADATA_KEY_PREFIX);
+        properties.add(CONTENT_BUFFER_SIZE);
+        this.properties = Collections.unmodifiableList(properties);
+
+        final Set<Relationship> relationships = new HashSet<>();
+        relationships.add(SUCCESS);
+        relationships.add(FAILURE);
+        this.relationships = Collections.unmodifiableSet(relationships);
+    }
+
+    @Override
+    public Set<Relationship> getRelationships() {
+        return this.relationships;
+    }
+
+    @Override
+    protected List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return this.properties;
+    }
+
+    @SuppressWarnings("unused")
+    @OnScheduled
+    public void onScheduled(ProcessContext context) {
+        final String mimeTypeFilterInput = context.getProperty(MIME_TYPE_FILTER).getValue();
+        if (mimeTypeFilterInput != null && mimeTypeFilterInput.length() > 0) {
+            mimeTypeFilterRef.set(Pattern.compile(mimeTypeFilterInput));
+        } else {
+            mimeTypeFilterRef.set(null);
+        }
+
+        String metadataKeyFilterInput = context.getProperty(METADATA_KEY_FILTER).getValue();
+        if (metadataKeyFilterInput != null && metadataKeyFilterInput.length() > 0) {
+            metadataKeyFilterRef.set(Pattern.compile(metadataKeyFilterInput));
+        } else {
+            metadataKeyFilterRef.set(null);
+        }
+
+        autoDetectParser = new AutoDetectParser();
+    }
+
+    @Override
+    public void onTrigger(final ProcessContext context, final ProcessSession session) throws ProcessException {
+        FlowFile flowFile = session.get();
+        if (flowFile == null) {
+            return;
+        }
+
+        final ProcessorLog logger = this.getLogger();
+        final ObjectHolder<Map<String, String>> value = new ObjectHolder<>(null);
+        final Integer maxAttribCount = context.getProperty(MAX_NUMBER_OF_ATTRIBUTES).asInteger();
+        final Integer maxAttribLength = context.getProperty(MAX_ATTRIBUTE_LENGTH).asInteger();
+        final Integer contentBufferSize = context.getProperty(CONTENT_BUFFER_SIZE).evaluateAttributeExpressions().asInteger();
+        final String prefix = context.getProperty(METADATA_KEY_PREFIX).evaluateAttributeExpressions().getValue();
```

Comment:
`contentBufferSize` doesn't need EL support, so I'm removing that.

`prefix` fixed to use to `...evaluteAttributeExpressions(flowFile)...`.